common/views/components/ConsentAndScripts/index.tsx

```diff
@@ -0,0 +1,49 @@
+import Head from 'next/head';
+import { useEffect, useState } from 'react';
+import CivicUK from './ConsentAndScripts.CivicUK';
+import { getAnalyticsConsentState } from '@weco/common/services/app/civic-uk';
+
+const ConsentAndScripts = ({ segmentSnippet }: { segmentSnippet: string }) => {
+  const [consentState, setConsentState] = useState(false);
+
+  const onAnalyticsConsentChanged = (event: CustomEvent) => {
+    // Toggle rendering of scripts
+    setConsentState(event.detail.consent === 'granted');
+
+    // Update datalayer config with consent value
+    gtag('consent', 'update', {
+      ad_storage: event.detail.consent,
+      analytics_storage: event.detail.consent,
+    });
+  };
+
+  useEffect(() => {
+    setConsentState(getAnalyticsConsentState());
+
+    window.addEventListener(
+      'analyticsConsentChanged',
+      onAnalyticsConsentChanged
+    );
+
+    return () => {
+      window.removeEventListener(
+        'analyticsConsentChanged',
+        onAnalyticsConsentChanged
+      );
+    };
+  }, []);
+
+  return (
+    <>
+      {consentState && (
+        <Head>
+          <script dangerouslySetInnerHTML={{ __html: segmentSnippet }} />
```

Comment:
I originally also put the GTM script in this conditional, but I was experiencing what I think is [this Next bug](https://github.com/vercel/next.js/issues/17085) where on loading it/removing it/re-adding it, random meta tags would duplicate themselves, even if they had a `key`. Someone mentioned that [for them it only happened when GTM was enabled](https://github.com/vercel/next.js/issues/17085#issuecomment-829608632), and it does seem to be the case here as well.

I think it's ok though as we set the consent to `denied`, so no hits are being sent.

Reply:
> so no hits are being sent.

I believe the tag still "phones home" via what the docs describe as a "cookie-less ping". 

There is the possibly contentious case for adding the GTM script before consent that we still record some analytics data about these users in a legally compliant manner (so long as we set the consent state in the data layer correctly). See https://support.google.com/tagmanager/answer/13802165?sjid=2210990408479536825-EU for more details about what gets communicated to Google in the situation consent is denied.

See "[Behavioral modeling for consent mode](https://support.google.com/analytics/answer/11161109?sjid=2210990408479536825-EU)" for one of the "features" available to us. Notably modelled data for un-consented visitors is not sent via any BigQuery integration.

We should be sure we're comfortable with this, and check with @LaurenFBaily.

Reply:
Had a look at this and the documentation, I'm happy with that.